packages/taco/src/sign.ts

```diff
@@ -0,0 +1,59 @@
+import { Domain, PorterClient } from '@nucypher/shared';
+
+import { SigningOptions, SignResult, UserOperation } from './types';
+
+// Core signing functions
+export async function sign191(
+  payload: Uint8Array | string,
+  cohortId: number,
+  domain: Domain,
+  options: SigningOptions = { optimistic: true, returnAggregated: true }
+): Promise<SignResult> {
```

Comment:
You also need to take `porterUris?: string[]`  - this is how we ensure that any porter instance can be used and not just the defaults for the domain.

See https://github.com/nucypher/taco-web/blob/main/packages/taco/src/taco.ts#L140-L145

Same for other sign API functions below.